pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationService.java

```diff
@@ -84,17 +85,21 @@ public AuthenticationService(ServiceConfiguration conf) throws PulsarServerExcep
         }
     }
 
-    public String authenticateHttpRequest(HttpServletRequest request) throws AuthenticationException {
+    public String authenticateHttpRequest(HttpServletRequest request, AuthenticationDataSource authData)
+            throws AuthenticationException {
         AuthenticationException authenticationException = null;
-        AuthenticationDataSource authData = new AuthenticationDataHttps(request);
-        String authMethodName = request.getHeader("X-Pulsar-Auth-Method-Name");
+        String authMethodName = request.getHeader(AuthenticationFilter.PULSAR_AUTH_METHOD_NAME);
 
         if (authMethodName != null) {
             AuthenticationProvider providerToUse = providers.get(authMethodName);
             if (providerToUse == null) {
                 throw new AuthenticationException(
                         String.format("Unsupported authentication method: [%s].", authMethodName));
             }
+            if (authData == null) {
+                AuthenticationState authenticationState = providerToUse.newHttpAuthState(request);
```

Comment:
`newHttpAuthState` throws `AuthenticationException`, especially for a class like the `TokenAuthenticationHttpState` because it calls `provider.authenticate` in the constructor. We should wrap it in a `try` block. Also, wouldn't it work to use the `getAuthRole()` method on the `authenticationState` to bypass a second authentication call?